include/sliding_window/search/search_setup.hpp

```diff
@@ -131,16 +134,20 @@ std::set<size_t> find_pattern_bins(pattern_bounds const & pattern, size_t const
 // TODO: pass slice of records directly instead of start, end and records
 //-----------------------------
 template <typename ibf_t, typename rec_vec_t>
-auto worker(size_t const start, size_t const end, ibf_t const & ibf, search_arguments const & arguments, 
-                                                                    rec_vec_t const & records, threshold const & threshold_data)
+std::vector<query_result> worker(size_t const start, 
```

Comment:
@marehr  I didn't refactor the `worker` since you mentioned that you can create a pull request from stellar

Reply:
As I said it is fine as it is :)